packages/cli-platform-apple/src/commands/buildCommand/buildProject.ts

```diff
@@ -11,15 +11,28 @@ import {
   getLoader,
 } from '@react-native-community/cli-tools';
 import type {BuildFlags} from './buildOptions';
+import {simulatorDestinationMap} from './simulatorDestinationMap';
 
 export function buildProject(
   xcodeProject: IOSProjectInfo,
+  platform: string,
   udid: string | undefined,
   mode: string,
   scheme: string,
   args: BuildFlags,
 ): Promise<string> {
   return new Promise((resolve, reject) => {
+    const simulatorDest = simulatorDestinationMap?.[platform];
+
+    if (!simulatorDest) {
+      reject(
+        new CLIError(
+          `Unknown platform: ${platform}. Please, use one of: ios, macos, visionos, tvos.`,
```

Comment:
nit: maybe let's move `ios, macos, visionos, tvos` to some constant so whenever in future we'll have a new platform, it will easier to refactor